### PR TITLE
Adds classes and dimensions to shortcode arguments

### PR DESCRIPTION
**Pull request by andrewdisley**
Recently made use of this plugin but I needed to add classes and dimensions on a per icon basis for the shortcode to allow for responsive sizing without having to wrap a responsive container around each SVG.

Previously wrapping element around the shortcode to achieve the desired result worked but in certain instances was causing FOUC and CLS.

Previously:
`<div class="h-6 w-6 lg:h-9 lg:w-9">{% heroicon "solid", "check" %}</div>`

Now:
`{% heroicon "solid", "arrow-sm-left", "", "h-6 mr-1 w-6", "24" %}`

The only potentially confusing part now is the need to include the empty string for the alt attribute, certainly welcome an improvement here to make it clearer. If this change is out of scope for your roadmap just disregard.

### CODE REVIEW

**Comment by SeanMcP** [comment]
Hey! I like the idea of adding the ability to customize more about the markup, but I'm hesitant to add more ordered parameters. What do you think of a params object?

```html
{% heroicon "outline", "check", { alt: "", className: "h-6 mr-1 w-6", width: 24 } %}
```

If that sounds reasonable, I'm down to prepare a breaking release with the new API.

**Comment by andrewdisley** [comment]
That is a great idea Sean. If the new params object is an attributes map that would make it really extensible.

`{% heroicon "outline", "check", { aria-hidden: "true", class: "h-6 mr-1 w-6", width: 24, x-show: "open" } %}`

**Comment by SeanMcP** [comment]
Oo, I like the ability to add Alpine attributes! I might block some attributes like `aria-hidden` and `aria-label` since the library handles that with the `alt` parameter, but everything else should be fair game.

If you're interested in trying to implement the params object, you are more than welcome! If not, I will put something together in the coming days.

File: test/runs/nunjucks.html [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
-<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" fill="currentColor" aria-hidden="true" data-heroicon-name="archive" data-heroicon-style="solid"><path d="M4 3a2 2 0 100 4h12a2 2 0 100-4H4z"/><path fill-rule="evenodd" d="M3 8h14v7a2 2 0 01-2 2H5a2 2 0 01-2-2V8zm5 3a1 1 0 011-1h2a1 1 0 110 2H9a1 1 0 01-1-1z" clip-rule="evenodd"/></svg>
-<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" fill="currentColor" data-heroicon-name="archive" data-heroicon-style="solid"><title>with an alt tag</title><path d="M4 3a2 2 0 100 4h12a2 2 0 100-4H4z"/><path fill-rule="evenodd" d="M3 8h14v7a2 2 0 01-2 2H5a2 2 0 01-2-2V8zm5 3a1 1 0 011-1h2a1 1 0 110 2H9a1 1 0 01-1-1z" clip-rule="evenodd"/></svg>
+<svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" aria-hidden="true" class="undefined" data-heroicon-name="archive" data-heroicon-style="outline"><path stroke-linecap="round" stroke-linejoin="round" d="M5 8h14M5 8a2 2 0 110-4h14a2 2 0 110 4M5 8v10a2 2 0 002 2h10a2 2 0 002-2V8m-9 4h4"/></svg>
```
Comment (SeanMcP) [inline test/runs/nunjucks.html:4]: Ideally we don't want `class="undefined"` applied to the `svg`.

**Comment by flortsch** [comment]
This would be a nice addition :)

**Comment by SeanMcP** [comment]
Uh oh! Me putting something together in the "coming days" turned into... forgetting. 😬

Thanks for the bump @flortsch!

**Comment by SeanMcP** [comment]
@andrewdisley @flortsch I have opened #5 if either of you have time for a review.

**Comment by andrewdisley** [comment]
@SeanMcP Finally got around to reviewing #5 and updating our usage and all is working as expected.

#5 is spot on and supersedes the need for #4 - closing.

**Pull request closed without merging**